src/p2p/net_node.inl

```diff
@@ -2288,13 +2285,15 @@ namespace nodetool
     }
 
     //fill response
-    rsp.local_time = time(NULL);
-
     const epee::net_utils::zone zone_type = context.m_remote_address.get_zone();
     network_zone& zone = m_network_zones.at(zone_type);
 
-    zone.m_peerlist.get_peerlist_head(rsp.local_peerlist_new, true);
+    zone.m_peerlist.get_peerlist_head(rsp.local_peerlist_new, true, P2P_DEFAULT_PEERS_IN_HANDSHAKE, [&context](const peerlist_entry &pe) {
+      return context.sent_addresses.find(pe.adr) == context.sent_addresses.end();
+    });
     m_payload_handler.get_payload_sync_data(rsp.payload_data);
+    for (const auto &e: rsp.local_peerlist_new)
+      context.sent_addresses.insert(e.adr);
```

Comment:
The idea is that eventually we stop sending peerlists ... ? Interesting because this could reveal when someone _just_ moved into your whitelist, and thus potentially leaking current connections. However, the privacy leak isn't guaranteed because the p2p code will probe the greylist over time and not make a permanent connection. Constantly sending random peers mitigates any leak.

I'm not immediately aware of a major privacy issue - Dandelion++ fluff stage should "break-up" any analysis of knowing the network graph. I (may have) incorrectly stated somewhere that we may need to "fluff" blocks in Monero - if it were true, than this change could be bad. I need to read the dandelion/dandelion++ to confirm this though - I forgot how they analyze their protocol which determines the importance of keeping the network graph concealed.

I'm primarily listing this to make everyone aware of any subtle changes here, and make sure they aren't a surprise to all.

Reply:
Yes. We'd be sending updates only to long connected peers. I guess we could make it so get_peerlist_head does its usual thing, then the caller calculates the delta and sends it. This would fix that possible vector.

Reply:
Which I did.

Reply:
Another consequence of this sytem:
Currently, when we bump an address off our grey list as it fails to connect, we're pretty likely to get it re-relayed to us by a peer, so it gets back into the grey list. With this change, it becomes much less likely.